pkg/web_app/lib/src/widget/downloads_chart/computations.dart

```diff
@@ -41,3 +41,40 @@ Iterable<String> prepareRanges(List<VersionRangeCount> rangeDownloads) {
 
   return (ranges: ranges, weekLists: result.reversed.toList());
 }
+
+/// Calculates closest point from [point] to line defined by [startPoint] and
+/// [endPoint].
+///
+/// This assumes that the closest point is on the line, that is within the
+/// two endpoints. Returns `(double.maxFinite, double.maxFinite)` if this is
+/// not the case.
+(num, num) closestPointOnLine(
+    (num, num) startPoint, (num, num) endPoint, (num, num) point) {
+  final directionVector =
```

Comment:
Nit, perhaps this should just be 'd' or 'direction'

Reply:
done